sklearn/datasets/samples_generator.py

```diff
@@ -613,20 +613,23 @@ def make_circles(n_samples=100, shuffle=True, noise=None, random_state=None,
 
     if factor > 1 or factor < 0:
         raise ValueError("'factor' has to be between 0 and 1.")
+        
+    n_samples_out = n_samples // 2
+    n_samples_in = n_samples - n_samples_out
 
     generator = check_random_state(random_state)
-    # so as not to have the first point = last point, we add one and then
-    # remove it.
-    linspace = np.linspace(0, 2 * np.pi, n_samples // 2 + 1)[:-1]
-    outer_circ_x = np.cos(linspace)
-    outer_circ_y = np.sin(linspace)
-    inner_circ_x = outer_circ_x * factor
-    inner_circ_y = outer_circ_y * factor
+    # so as not to have the first point = last point, we set endpoint=False
+    linspace_out = np.linspace(0, 2 * np.pi, n_samples_out, endpoint=False)
+    linspace_in = np.linspace(0, 2 * np.pi, n_samples_in, endpoint=True)
```

Comment:
Why `endpoint=True` in the second?

Reply:
Thanks. I feel like I thought it might have been necessary, but I cannot think of a proper reason right now. I'll chnage it to ```endpoint=False```. Thanks.